### ISSUE TITLE
Editor should be focusable in read only mode

### ISSUE THREAD

**Issue by scofalik** [ckeditor/ckeditor5#6080]
## 📝 Provide a description of the improvement

At the moment, when we switch the editor to the read-only mode, the editor cannot be focused anymore. As I wonder about it, it doesn't make much sense. You still can click around (and the selection moves). You can click, for example, on a comment marker to highlight a comment. I think that those two properties - focusing and being editable - should not be connected. They mean different things.

Some features behaviors are based on focusing and blurring the editor so it would be nice to have focus working consistently both in the editing mode and in the read-only mode.

My case is that, in the read-only mode, in comments feature, after fixing some other bugs, I have new problems and correct focus handling might solve them.

**Comment by mlewand** [comment]
#6200 is a duplicate of this issue, but since the former has more discussion and a related PR open already I'll close this one.
